packages/block-library/src/latest-posts/edit.js

```diff
@@ -27,15 +29,33 @@ import {
 } from '@wordpress/editor';
 import { withSelect } from '@wordpress/data';
 
+/**
+ * Module Constants
+ */
+const CATEGORIES_LIST_QUERY = {
+	per_page: 100,
+};
 const MAX_POSTS_COLUMNS = 6;
 
 class LatestPostsEdit extends Component {
 	constructor() {
 		super( ...arguments );
-
+		this.state = {
+			categoriesList: [],
+		};
 		this.toggleDisplayPostDate = this.toggleDisplayPostDate.bind( this );
 	}
 
+	componentWillMount() {
+		apiFetch( {
+			path: addQueryArgs( `/wp/v2/categories`, CATEGORIES_LIST_QUERY ),
+		} ).then(
+			( categoriesList ) => {
+				this.setState( { categoriesList } );
```

Comment:
What if the block is removed at this point, we should guard against it

Reply:
Added an abort call when the component unmounts, same as the hierarchical term selector.